### DISCUSSION TITLE
What dose the parameter "concurrency_start_early_phase" mean ?

### DISCUSSION THREAD

**Discussion by phystry** [swar/Swar-Chia-Plot-Manager#428] [Q&A]
# concurrency_start_early_phase: The phase in which you want to start a plot early. It is recommended to use 4 for this field.
To my understanding , if I want the early-started task starts at phase 4, I need to set this parameter to 4. I did so, but the early-started task starts at phase 2.

**Comment by dJani97** [comment]
If you have `max_concurrent` set to anything bigger than 1, then a new task will start automatically as soon as the `stagger_minutes` expires.
As far as I understand, `concurrency_start_early_phase` can override the current `max_concurrent` setting, and start a task early even if the number of currently running tasks has already reached `max_concurrent`.

**Comment by swar** [comment]
@dJani97  almost has it.

It's not when stagger minute expires, its when one job enters another phase you set in the `concurrency_start_early_phase`

So if I have
`max_concurrent` = 2
`max_concurrent_with_start_early` = 3
`concurrency_start_early_phase` = 4
`concurrency_start_early_phase_delay` = 0

Then if I have 2 plots currently running, then if one plot hits phase 4. The manager will kick off another job. It's usefulness lies in the fact that phase 4 is very quick and moving a plot to a destination drive doesn't take all that long. So you can get a head start on another plot while your system has available resources.

**Comment by fabiomesmu** [comment]
> # concurrency_start_early_phase: A fase em que você deseja iniciar um gráfico antecipadamente. Recomenda-se usar 4 para este campo.
> No meu entendimento, se eu quiser que a tarefa iniciada antecipadamente comece na fase 4, preciso definir esse parâmetro como 4. Fiz isso, mas a tarefa iniciada antecipadamente começa na fase 2.

Pensei a mesma coisa, e era esta a intenção mesmo!
